components/scream/src/control/surface_coupling.cpp

```diff
@@ -324,6 +459,11 @@ get_col_info(const std::shared_ptr<const FieldHeader>& fh,
     me = p;
     p = me->get_parent().lock();
   }
+
+  // Surface corresponds to k=nlevs-1
+  // Hacky, but need to ask Luca about vector3d
+  if (lt==LayoutType::Scalar3D)
+    col_offset+= m_num_levs-1;
```

Comment:
Yeah, we need vector3d support. Should be fairly easy to do though, right?

Reply:
I'm actually looking at how col_offset is computed, and I am not 100% sure it is correct. In particular, for vector fields, it's init-ed to the vector dimension, I believe because I was thinking that the surface was the 1st entry in a column. Would you mind double checking that logic?

Reply:
Yeah, I'll go through it.